An error on `osx-arm64`:
```
FAILED: libmamba/CMakeFiles/libmamba.dir/src/solv-cpp/solver.cpp.o 
/conda-bld/mamba-split_1694082368697/_build_env/bin/arm64-apple-darwin20.0.0-clang++ -DFMT_SHARED -DLIBMAMBA_EXPORTS -DSPDLOG_FMT_EXTERNAL -Dlibmamba_EXPORTS -I/conda-bld/mamba-split_1694082368697/work/libmamba/include -I/conda-bld/mamba-split_1694082368697/work/libmamba/src -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /conda-bld/mamba-split_1694082368697/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include -fdebug-prefix-map=/conda-bld/mamba-split_1694082368697/work=/usr/local/src/conda/libmamba-1.5.1 -fdebug-prefix-map=/conda-bld/mamba-split_1694082368697/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh=/usr/local/src/conda-prefix -D_LIBCPP_DISABLE_AVAILABILITY=1 -std=c++17 -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=11.0 -fPIC -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wold-style-cast -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wconversion -Wsign-conversion -Wnull-dereference -Wdouble-promotion -Wformat=2 -Wunreachable-code -Wuninitialized -MD -MT libmamba/CMakeFiles/libmamba.dir/src/solv-cpp/solver.cpp.o -MF libmamba/CMakeFiles/libmamba.dir/src/solv-cpp/solver.cpp.o.d -o libmamba/CMakeFiles/libmamba.dir/src/solv-cpp/solver.cpp.o -c /conda-bld/mamba-split_1694082368697/work/libmamba/src/solv-cpp/solver.cpp
/conda-bld/mamba-split_1694082368697/work/libmamba/src/solv-cpp/solver.cpp:235:18: error: no member named 'solver_ruleinfo2str' in the global namespace
        return ::solver_ruleinfo2str(
               ~~^
1 error generated.
```

@jezdez Seems like the error was caused by this https://github.com/mamba-org/mamba/blame/69a2cd30e2e17efc6df0f5d458ce5540cf8d40ad/libmamba/src/solv-cpp/solver.cpp#L235 (the upstream PR https://github.com/mamba-org/mamba/pull/2504/files)

--- a/recipe/build_mamba.bat
+++ b/recipe/build_mamba.bat
@@ -5,7 +5,7 @@ echo "Building %PKG_NAME%."
 
 if /I "%PKG_NAME%" == "mamba" (
 	cd mamba
-	%PYTHON% -m pip install . --no-deps -vv
+	%PYTHON% -m pip install . --no-deps --no-build-isolation -vv
 	exit 0
 )
 
@@ -14,7 +14,8 @@ mkdir build
 cd build
 if errorlevel 1 exit /b 1
 
-set "CXXFLAGS=%CXXFLAGS% /D_LIBCPP_DISABLE_AVAILABILITY=1"
+rem most likely don't needed on Windows, just for OSX
+rem set "CXXFLAGS=%CXXFLAGS% /D_LIBCPP_DISABLE_AVAILABILITY=1"
 
 :: Generate the build files.
 echo "Generating the build files..."
@@ -55,7 +56,7 @@ if errorlevel 1 exit /b 1
 if /I "%PKG_NAME%" == "libmambapy" (
 	cd ../libmambapy
 	rmdir /Q /S build
-	%PYTHON% -m pip install . --no-deps -vv
+	%PYTHON% -m pip install . --no-deps --no-build-isolation -vv
 	del *.pyc /a /s
 )
 

--- a/recipe/build_mamba.sh
+++ b/recipe/build_mamba.sh
@@ -2,12 +2,12 @@
 
 if [[ $PKG_NAME == "mamba" ]]; then
     cd mamba || exit 1
-    $PYTHON -m pip install . --no-deps -vv
-
+    $PYTHON -m pip install . --no-deps --no-build-isolation -vv
+    
     echo "Adding link to mamba into condabin";
     mkdir -p $PREFIX/condabin
     ln -s $PREFIX/bin/mamba $PREFIX/condabin/mamba
-
+    
     exit 0
 fi
 
@@ -22,22 +22,22 @@ echo "Generating the build files..."
 
 if [[ $PKG_NAME == "libmamba" ]]; then
     cmake .. ${CMAKE_ARGS}              \
-        -GNinja                         \
-        -DCMAKE_INSTALL_PREFIX=$PREFIX  \
-        -DCMAKE_PREFIX_PATH=$PREFIX     \
-        -DCMAKE_BUILD_TYPE=Release      \
-        -DBUILD_LIBMAMBA=ON             \
-        -DBUILD_SHARED=ON               \
-        -DBUILD_MAMBA_PACKAGE=ON
-elif [[ $PKG_NAME == "libmambapy" ]]; then
+    -GNinja                         \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX  \
+    -DCMAKE_PREFIX_PATH=$PREFIX     \
+    -DCMAKE_BUILD_TYPE=Release      \
+    -DBUILD_LIBMAMBA=ON             \
+    -DBUILD_SHARED=ON               \
+    -DBUILD_MAMBA_PACKAGE=ON
+    elif [[ $PKG_NAME == "libmambapy" ]]; then
     # TODO finds wrong python interpreter!!!!
     cmake .. ${CMAKE_ARGS}              \
-        -GNinja                         \
-        -DCMAKE_PREFIX_PATH=$PREFIX     \
-        -DCMAKE_INSTALL_PREFIX=$PREFIX  \
-        -DCMAKE_BUILD_TYPE=Release      \
-        -DPython_EXECUTABLE=$PYTHON     \
-        -DBUILD_LIBMAMBAPY=ON
+    -GNinja                         \
+    -DCMAKE_PREFIX_PATH=$PREFIX     \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX  \
+    -DCMAKE_BUILD_TYPE=Release      \
+    -DPython_EXECUTABLE=$PYTHON     \
+    -DBUILD_LIBMAMBAPY=ON
 fi
 
 # Build.
@@ -51,7 +51,7 @@ ninja install || exit 1
 if [[ $PKG_NAME == "libmambapy" ]]; then
     cd ../libmambapy || exit 1
     rm -rf build
-    $PYTHON -m pip install . --no-deps -vv
+    $PYTHON -m pip install . --no-deps --no-build-isolation -vv
     find libmambapy/bindings* -type f -print0 | xargs -0 rm -f --
 fi
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -35,7 +35,7 @@ outputs:
         - ninja
         - python            # [win]
       host:
-        - libsolv
+        - libsolv 0.7.24
         - libcurl
         - openssl {{ openssl }}
         - libarchive
@@ -51,7 +51,7 @@ outputs:
         - bzip2
         - cli11
       run:
-        - libsolv
+        - libsolv >=0.7.24,<0.8.0a0
         - reproc-cpp
         - reproc
     test:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -35,25 +35,25 @@ outputs:
         - ninja
         - python            # [win]
       host:
-        - libsolv 0.7.22
-        - libcurl 7.88.1
+        - libsolv
+        - libcurl
         - openssl {{ openssl }}
-        - libarchive 3.6.2
-        - nlohmann_json 3.11.2
-        - cpp-expected 1.0.0
-        - reproc-cpp 14.2.4
-        - reproc 14.2.4
-        - spdlog 1.11.0
-        - yaml-cpp 0.7.0
-        - fmt 9.1.0
-        - winreg 4.1.2 # [win]
-        - zstd 1.5.4
-        - bzip2 1.0
-        - cli11 2.1.2
+        - libarchive
+        - nlohmann_json
+        - cpp-expected
+        - reproc-cpp
+        - reproc
+        - spdlog
+        - yaml-cpp
+        - fmt
+        - winreg  # [win]
+        - zstd
+        - bzip2
+        - cli11
       run:
-        - libsolv >=0.7.22,<0.8.0a0
-        - reproc-cpp >=14.2.4,<15.0a0
-        - reproc >=14.2.4,<15.0a0
+        - libsolv
+        - reproc-cpp
+        - reproc
     test:
       commands:
         - test -d ${PREFIX}/include/mamba                                   # [unix]
@@ -92,14 +92,14 @@ outputs:
         - pip
         - setuptools
         - wheel
-        - pybind11 2.10.1
-        - pybind11-abi 4
-        - openssl {{ openssl }}
-        - yaml-cpp 0.7.0
-        - cpp-expected 1.0.0
-        - spdlog 1.11.0
-        - fmt 9.1.0
-        - nlohmann_json 3.11.2
+        - pybind11
+        - pybind11-abi
+        - openssl
+        - yaml-cpp
+        - cpp-expected
+        - spdlog
+        - fmt
+        - nlohmann_json
         - {{ pin_subpackage('libmamba', exact=True) }}
       run:
         - python

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -47,8 +47,8 @@ outputs:
         - yaml-cpp
         - fmt
         - winreg  # [win]
-        - zstd
-        - bzip2
+        - zstd {{ zstd }}
+        - bzip2 {{ bzip2 }}
         - cli11
       run:
         - libsolv >=0.7.24,<0.8.0a0

